app/controllers/schools_controller.rb

```diff
@@ -0,0 +1,38 @@
+class SchoolsController < ApplicationController
+  def show
+    @school = School.find(params[:id])
+  end
+
+  def new; end
+
+  def create
+    validate_opened_date
+    return if performed?
+
+    @school = School.new(params.require(:school).permit(:name, :opened, :school_type))
+
+    if @school&.save
+      redirect_to @school
+    else
+      render "new"
+    end
+  end
+
+  private
+  def validate_opened_date
+    unless Date.valid_date?(*opened_date_parts)
+      @school = School.new
+      @school.errors.add(:opened, 'date validation error text goes here')
+      render "new"
```

Comment:
Probably not the pattern we want to stick with - I am pretty sure this will wipe all the user-entered values.

I think a better idea would be to always make a new school in `create` and add a validator to the model checking for date being correct. Arguably that's where validation belongs in MVC anyway :) 

As a side benefit it will let you get rid of
```
return if performed?
```
line.